app/scripts/controllers/dashboard.coffee

```diff
@@ -326,8 +326,15 @@ angular.module('edudashAppCtrl').controller 'DashboardCtrl', [
           loadingSrv.containerLoad schoolsP, document.getElementById mapId
           schoolsP.then (schools) ->
             if $scope.selected? and $scope.viewMode == 'schools'
+              $scope.selectedCodeYear =
+                code: $scope.selected.CODE
+                year: $scope.year
               $scope.select $scope.selected.CODE
 
+        $scope.$watch 'selectedCodeYear', (selectedCodeYear) -> if selectedCodeYear?
+          (findSchool selectedCodeYear.code).then (school)->
+            $scope.selected = school
```

Comment:
yes, that is was the reason It works when we talk. my bad. We need to figure out how to implement it.